x/payment/types/builder.go

```diff
@@ -112,14 +137,15 @@ func (k *KeyringSigner) BuildSignedTx(builder sdkclient.TxBuilder, msg sdktypes.
 		return nil, err
 	}
 
+	signerData, err := k.GetSignerData()
+	if err != nil {
+		return nil, err
+	}
+
 	// Generate the bytes to be signed.
 	bytesToSign, err := k.encCfg.TxConfig.SignModeHandler().GetSignBytes(
 		signing.SignMode_SIGN_MODE_DIRECT,
-		authsigning.SignerData{
-			ChainID:       k.chainID,
-			AccountNumber: accountNumber,
-			Sequence:      sequence,
-		},
+		signerData,
```

Comment:
this was another bug where we weren't adding the public key and address to the `SignerData`, so the sign bytes were sometimes being generated incorrectly.